src/main/scala/it/pagopa/interop/agreementprocess/service/CatalogManagementService.scala

```diff
@@ -24,22 +24,52 @@ object CatalogManagementService {
   def validateCreationOnDescriptor(eservice: EService, descriptorId: UUID): Future[Unit]   = {
     val allowedStatus: List[EServiceDescriptorState] =
       List(EServiceDescriptorState.PUBLISHED, EServiceDescriptorState.SUSPENDED)
-    validateEServiceDescriptorStatus(eservice, descriptorId, allowedStatus)
+    validateEServiceDescriptorStatus(eservice, descriptorId, allowedStatus, true)
   }
   def validateSubmitOnDescriptor(eservice: EService, descriptorId: UUID): Future[Unit]     = {
     val allowedStatus: List[EServiceDescriptorState] =
       List(EServiceDescriptorState.PUBLISHED, EServiceDescriptorState.SUSPENDED)
-    validateEServiceDescriptorStatus(eservice, descriptorId, allowedStatus)
+    validateEServiceDescriptorStatus(eservice, descriptorId, allowedStatus, true)
   }
 
```

Comment:
I suggest a different approach.

We can create two functions, each one solving a specific problem. Both of them will leverage on a third function for the common state verification logic:

```
 def validateLatestDescriptor(
    eService: EService,
    descriptorId: UUID,
    allowedStates: List[EServiceDescriptorState]
  ): Future[Unit] =
      eService.descriptors
        .filterNot(_.state == EServiceDescriptorState.DRAFT)
        .maxByOption(_.version.toLong)
        .find(_.id == descriptorId)
        .toRight(NotLatestEServiceDescriptor(descriptorId))
        .flatMap(validateDescriptorState(d => eService.id, d.id, d.state, allowedStates))
        .toFuture

def validateDescriptor(eService: EService, descriptorId: UUID, allowedStates: List[EServiceDescriptorState]) = 
    eService.descriptors
        .find(_.id == descriptorId)
        .toRight(DescriptorNotFound(eService.id, descriptorId))
        .flatMap(validateDescriptorState(d => eService.id, d.id, d.state, allowedStates))
        .toFuture

def validateDescriptorState(eServiceId: UUID, descriptorId: UUID, descriptorState: EServiceDescriptorState, allowedStates: List[EServiceDescriptorState]) = 
    Either
      .left[DescriptorNotInExpectedState, Unit](DescriptorNotInExpectedState(eServiceId, descriptorId, allowedStates))
      .unlessA(allowedStates.contains(descriptorState))
```

(not sure this 100% compiles, I just wrote it here without the IDE)

And then you can use the specific functions for `validate*OnDescriptor`

Reasons:
- when possible, avoid `Boolean` parameters, they make the code less readable
- generally, if possible avoid default parameters
- code related to the status check was duplicated
- it is not required to filter by `DRAFT` for non-latest check, the given allowed states are enough